op-node/sources/eth_client.go

```diff
@@ -224,7 +273,7 @@ func (s *EthClient) BlockIDRange(ctx context.Context, begin eth.BlockID, max uin
 			if result == nil {
 				break // no more headers from here
 			}
-			info, err := result.Info(s.trustRPC)
+			info, err := result.Info(s.trustRPC, s.mustBePostMerge)
```

Comment:
I think its slightly ugly to pass in 2 args that are static properties to `hdr.Info`, I feel like having a method on the `EthClient` that you pass a `hdr` to that is optionally used before inserting an item into the cache or returning remote data to the caller would feel cleaner

Reply:
Agreed, although I'm planning to remove this `BlockIDRange` method since it's unused, so I think I'll keep this as is and open a separate PR to remove the method.